src/main/kotlin/fi/metatavu/plastep/productinformation/machines/MachinesApiImpl.kt

```diff
@@ -0,0 +1,31 @@
+package fi.metatavu.plastep.productinformation.machines
+
+import fi.metatavu.plastep.productinformation.lemon.LemonMachinesController
+import fi.metatavu.plastep.productinformation.rest.AbstractApi
+import fi.metatavu.plastep.productinformation.spec.MachinesApi
+import javax.enterprise.context.RequestScoped
+import javax.inject.Inject
+import javax.ws.rs.core.Response
+
+@RequestScoped
+class MachinesApiImpl : MachinesApi, AbstractApi() {
+
+    @Inject
+    lateinit var lemonMachinesController: LemonMachinesController
+
+    @Inject
+    lateinit var lemonMachineTranslator: LemonMachineTranslator
+
+    override fun listMachines(page: Int?, pageSize: Int?): Response {
+        if (!hasIntegrationRole()) {
+            return createForbidden("Only integration users can access this resource")
+        }
+
+        val lemonMachines = lemonMachinesController.listMachines(
+            page ?: 0,
+            pageSize ?: 10
```

Comment:
Maybe increase this limit to something like 50?

Reply:
Raised the limit, 10 was the same limit that the product listing has.